internal/models/models.go

```diff
@@ -52,10 +52,17 @@ func (p Publisher) AfterSave(tx *gorm.DB) error {
 	return nil
 }
 
+func (CodeHosting) TableName() string {
+	return "publishers_code_hosting"
+}
+
 type CodeHosting struct {
-	gorm.Model
-	URL         string `json:"url" gorm:"not null"`
-	PublisherID string `json:"publisherId"`
+	ID          string         `json:"-" gorm:"primarykey"`
```

Comment:
```suggestion
	ID          string         `json:"-" gorm:"primaryKey"`
```

This allowed [this code not to fail](https://github.com/italia/developers-italia-api/pull/125/commits/588e5d98237ecc5730bed7a4e351e970ca79b112#diff-4fda96c0894229f0095f701a008077c3520ebb6533e9816b61f7c75ebf8c7059L100) by creating `CodeHosting`s with the same zero value for ID (`""`).

Reply:
Nice catch, thanks!!!

Reply:
Looks like also other models need to be updated.
Do you want to update in this PR?